mudslide/models.py

```diff
@@ -553,12 +562,205 @@ def dV(self, X: ArrayLike) -> ArrayLike:
+
+
+            read_start = np.argwhere(np.array(["$coord\n" in line for line in lines]))[0][0]
+
+        with open(self.coord_path, "w") as coord_file:
+            current_line = read_start + 1
+
+            for i, coord in enumerate(X):  
+                if coord[0] < 0:
+                    s1 = first_space_neg
+                else:
+                    s1 = first_space
+
+                if coord[1] < 0:
+                    s2 = space_neg
+                else:
+                    s2 = space
+
+                if coord[2] < 0:
+                    s3 = space_neg
+                else:
+                    s3 = space
+                lines[current_line] = s1 + f"{coord[0]:1.14f}" + s2 + f"{coord[1]:1.14f}" + s3 + f"{coord[2]:1.14f}" + space + self.atom_order[i] + "\n"  
+                current_line+=1
+            coord_file.write("".join(lines))            
+
+
+    def call_and_parse_turbomole(self, outname="turbo.out"):
+        with open(outname, "w") as f:
+            ridft = subprocess.run("ridft", stdout=f)
+            rdgrad = subprocess.run("rdgrad", stdout=f)
+            egrad = subprocess.run("egrad", stdout=f)
+    
+        with open(outname, "r") as f:
+            data_dict = turboparse.parse_turbo(f)
+
+        parsed_nac_coupling = data_dict["egrad"]["coupling"]
+        
+        self.derivative_coupling = np.zeros((self.nstates(), self.nstates(), self.ndim()))
+        for i in range(self.nstates()):
+            for j in range(self.nstates()):
+                if i != j:
+                    for dct in parsed_nac_coupling:
+                        if (dct["bra_state"] == i) and (dct["ket_state"] == j):
+                            self.derivative_coupling[i][j] = np.array(dct["d/dR"]).reshape(self.ndim(), order="F")
+                            self.derivative_coupling[j][i] = self.derivative_coupling[i][j]
+
+        parsed_gradients = data_dict["rdgrad"]["gradient"]
+        parsed_gradients.extend(data_dict["egrad"]["gradient"])
+
+
+        self.gradients = np.zeros((self.nstates(),self.ndim()))
+        for state in self.states:
+            grads = []
+            for i in range(self.ndim() // 3):
+                grads.extend(
+                [
+                parsed_gradients[state]["d_dx"][i],
+                parsed_gradients[state]["d_dy"][i],
+                parsed_gradients[state]["d_dz"][i],
+                ]
+                        )
+            grads = np.array(grads)
+            self.gradients[state] = grads
+
+
+        self.force = -(self.gradients) 
+
+        parsed_energies = data_dict["egrad"]["excited_state"][0]["energy"]
+        energy = data_dict["ridft"]["energy"]
+        excited_energies = [
+            data_dict["egrad"]["excited_state"][i]["energy"] + energy for i in range(len(data_dict["egrad"]["excited_state"]))
+        ]
+
+        self.energies = [energy]
+        self.energies.extend(excited_energies)
+
+
+    def V(self, X: ArrayLike) -> ArrayLike:
```

Comment:
Why is this function necessary?

Reply:
Well if I don't use this function I get this error - and this is I think has to do with electronics.py - also it says "diabatic model" which is not true so maybe a typo? 

<img width="846" alt="Screen Shot 2022-04-07 at 4 50 49 PM" src="https://user-images.githubusercontent.com/46786940/162299593-98954ca2-3c96-41b9-8ae3-0c5e91573a27.png">

Reply:
Oh that's because you're deriving from `AdiabaticModel_` when you should be just driving from the `ElectronicsModel_` base class. `AdiabaticModel_` is for used for things like the Shin-Metiu model.